Sources/StreamVideo/CallStateMachine/Stages/Call+Stage.swift

```diff
@@ -0,0 +1,119 @@
+//
+// Copyright © 2025 Stream.io Inc. All rights reserved.
+//
+
+import Combine
+import Foundation
+
+extension Call.StateMachine {
+    /// Represents a stage in the call state machine.
+    class Stage: StreamStateMachineStage, @unchecked Sendable {
+
+        struct Context {
+            struct JoinInput: ReflectiveStringConvertible {
+                var create: Bool
+                var callSettings: CallSettings?
+                var options: CreateCallOptions?
+                var ring: Bool
+                var notify: Bool
+                var deliverySubject: PassthroughSubject<JoinCallResponse, Error>
+
+                var currentNumberOfRetries = 0
+                var maxNumberOfRetries = 3
+                var retryDelayRange: ClosedRange<TimeInterval> = 0.25...0.5
+            }
+
+            struct AcceptingInput {
+                var deliverySubject: PassthroughSubject<AcceptCallResponse, Error>
+            }
+
+            struct RejectingInput {
+                var reason: String?
+                var deliverySubject: PassthroughSubject<RejectCallResponse, Error>
+            }
+
+            weak var call: Call?
+
+            var joinInput: JoinInput? = nil
```

Comment:
Unsure a bit about all these optional variables, we now have everything in this context.

Reply:
Yes, that was intended. By having things in the context it's easier to follow and allows stages to be isolated.

Reply:
Not a huge fan of it, it's unclear whether all values can be `nil` or only one can be non-nil etc. Maybe we should represent this with enums? (Anyway, not a big thing)

Reply:
Great suggestion! Updated